explanation/dnssec/dnssec.md

```diff
@@ -0,0 +1,254 @@
+
+Given what we have:
+ * an application
+ * stub resolver ("dns client")
+ * recursive DNS server in the local network
+ * several other DNS servers in the internet that will be queried by our recursive DNS server
+
+Where does the DNSSEC validation happen? Who is responsible?
+
+Well, any DNS server can perform the validation. Let's look at two scenarios, and what it means in each case.
+
+### Validating Resolver
+
+When a recursive DNS server is also performing DNSSEC validation, it's called a *Validating Resolver*. That will typically be the DNS server on your local network, at your company, or in some cases at your ISP.
+
+![Validating Resolver](../images/ubuntu-local-validating-resolver.png)
+
+This is the case if you install the `bind9` DNS server: the default configuration is to act as a validating resolver. This can be seen in `/etc/bind/named.conf.options` after installing the `bind9` package:
+
+    options {
+        ...
+        dnssec-validation auto;
+        ...
+    };
+
+The key aspect of this deployment type is that the network between the stub resolver and the validating resolver has to be trusted, otherwise this is not useful. The Validating Resolver will be performing all the DNSSEC checks, and only returning a response if it was validated, but that response could have been altered on that "last mile" network segment.
+
+This is where the `trusted-ad` setting from `/etc/resolv.conf` comes into play:
+
+    nameserver 127.0.0.53
+    options edns0 trust-ad
+
+The `trust-ad` setting is documented in the [resolv.conf(5)](https://manpages.ubuntu.com/manpages/noble/man5/resolv.conf.5.html) manpage. It means that the local resolver will:
+
+ * Set the `ad` bit (Authenticated Data) in the outgoing queries.
+ * Trust the `ad` bit in the responses from the specified nameserver.
+
+When the `ad` bit is set in a DNS response, it means that DNSSEC validation was performed and successful. The data was authenticated.
+
+Specifyhing `trust-ad` in `/etc/resolv.conf` implies in these assumptions:
+
+ * The 127.0.0.53 name server is trusted to set the `ad` flag correctly in its responses. If it performs DNSSEC validation, it is trusted to perform this validation correctly, and set the `ad` flag accordingly. If it does not perform DNSSEC validation, then the `ad` flag will always be unset in the responses.
+ * The network path between localhost and 127.0.0.53 is trusted.
+
+When using systemd-resolved as the stub resolver, like in the config above, the network path is the localhost interface, and that is inherently trusted. But the actual nameserver that will be used is not 127.0.0.53: it will be whatever systemd-resolved was configured with. Unless local DNSSEC validation is enabled, systemd-resolved will remove the `ad` bit in the queries it forwards to the validating resolver, and also remove it from the responses.
```

Comment:
When using systemd-resolved as a stub resolver, as configured above, the network path to the local DNS resolver is inherently trusted, as it typically involves a localhost interface. However, the actual nameserver used might not be 127.0.0.53; it depends on systemd-resolved's configuration. Unless local DNSSEC validation is enabled, systemd-resolved will strip the ad bit from queries sent to the validating resolver and from the received responses.

Reply:
adopted slightly modified